src/resource/ResourceLoader.cpp

```diff
@@ -18,20 +18,22 @@ ResourceLoader::ResourceLoader(std::shared_ptr<Window> context) : mContext(conte
 }
 
 void ResourceLoader::RegisterGlobalResourceFactories() {
-    RegisterResourceFactory(ResourceType::Texture, std::make_shared<TextureFactory>());
-    RegisterResourceFactory(ResourceType::Vertex, std::make_shared<VertexFactory>());
-    RegisterResourceFactory(ResourceType::DisplayList, std::make_shared<DisplayListFactory>());
-    RegisterResourceFactory(ResourceType::Matrix, std::make_shared<MatrixFactory>());
-    RegisterResourceFactory(ResourceType::Array, std::make_shared<ArrayFactory>());
-    RegisterResourceFactory(ResourceType::Blob, std::make_shared<BlobFactory>());
+    RegisterResourceFactory(ResourceType::Texture, "Texture", std::make_shared<TextureFactory>());
```

Comment:
Why are we including the string name along with the resource type enum? Is it to match what comes out of the XML?

Reply:
Yup, that's how it checks which resource types are in the XML.